src/verifySignedMessage.js

```diff
@@ -0,0 +1,130 @@
+
+const ethToTokenParamTypes = [
+  { name: 'bitmapIndex', type: 'uint256' },
+  { name: 'bit', type: 'uint256'},
+  { name: 'token', type: 'address'},
+  { name: 'ethAmount', type: 'uint256'},
+  { name: 'tokenAmount', type: 'uint256'},
+  { name: 'expiryBlock', type: 'uint256'},
+  { name: 'to', type: 'address'},
+  { name: 'data', type: 'bytes'}
+]
+
+const tokenToEthParamTypes = [
+  { name: 'bitmapIndex', type: 'uint256' },
+  { name: 'bit', type: 'uint256'},
+  { name: 'token', type: 'address'},
+  { name: 'tokenAmount', type: 'uint256'},
+  { name: 'ethAmount', type: 'uint256'},
+  { name: 'expiryBlock', type: 'uint256'},
+  { name: 'to', type: 'address'},
+  { name: 'data', type: 'bytes'}
+]
+
+const tokenToTokenParamTypes = [
+  { name: 'bitmapIndex', type: 'uint256' },
+  { name: 'bit', type: 'uint256'},
+  { name: 'tokenIn', type: 'address'},
+  { name: 'tokenOut', type: 'address'},
+  { name: 'tokenInAmount', type: 'uint256'},
+  { name: 'tokenOutAmount', type: 'uint256'},
+  { name: 'expiryBlock', type: 'uint256'},
+  { name: 'to', type: 'address'},
+  { name: 'data', type: 'bytes'}
+]
+
+const verifySignedMessage = (signedMessage) => {
+  const { accountAddress, signature, message, signer } = signedMessage
+
+  if (!accountAddress) {
+    throw new Error('verifySignedMessage Error: Account Address not provided in signedMessage')
+  }
+
+  if (!signature) {
+    throw new Error('verifySignedMessage Error: Signature not provided in signedMessage')
+  }
+
+  if (!message) {
+    throw new Error('verifySignedMessage Error: Message not provided in signedMessage')
+  }
+
+  if (!signer) {
+    throw new Error('verifySignedMessage Error: Signer not provided in signedMessage')
+  }
+
+  const recoveredSigner = recoverSigner({ signature, typedDataHash: message})
+  if (recoveredSigner !== signer) {
+    throw new Error(`verifySignedMessage Error: Provided Signer ${signer} does not match Signer ${recoveredSigner} in Signed Message`)
+  }
+
+  const computedAccountAddress = accountFromOwner(signer)
+  if (toChecksumAddress(computedAccountAddress) !== toChecksumAddress(accountAddress)) {
+    throw new Error(`verifySignedMessage Error: Account Address ${accountAddress} does not match Computed Address ${computedAccountAddress}`)
+  }
+
+  const signedParams = signedMessage.signedParams
+  if (!signedParams || !signedParams[1]) {
+    throw new Error('verifySignedMessage Error: signedParams not provided in signedMessage')
+  }
+
+  const callData = signedMessage.signedParams[1].callData
+  if (!callData) {
+    throw new Error('verifySignedMessage Error: callData not provided in signedParams')
+  }
+
+  const functionName = callData.functionName
+  const params = callData.params
+  if (!functionName) {
+    throw new Error(`verifySignedMessage Error: No functionName specified in callData`)
+  }
+  if (!params) {
+    throw new Error(`verifySignedMessage Error: No params specified in callData`)
+  }
+  switch(functionName) {
```

Comment:
these are verifying that the params passed in match the provided function type, but we don't have any verification to check if the params passed in match the message/signature. If this were used in an API check, I could send a message that doesn't match the params, so executors might think it's executable when it's not. Wouldn't be an attack necessarily, but would be annoying for our executors and others.

I think what this should do is pass the incoming signed message values into this function https://github.com/brinktrade/brink-sdk/blob/7bd53f9e240055e910a9f8db52af776ac87acc0b/src/typedDataEIP712.js#L3 and check that the returned `typedDataHash` matches the `message`

If we do that check, we don't need specific parameter verifications (the `verifyParams` functions can be removed). I think we can remove the switch and specific function checks altogether. As long as the provided params match up with the message hash, this function can return that it's a valid signed message